apis/vshn/v1/dbaas_vshn_postgresql.go

```diff
@@ -280,3 +265,7 @@ type XVSHNPostgreSQLList struct {
 
 	Items []XVSHNPostgreSQL `json:"items"`
 }
+
+func (pg *VSHNPostgreSQL) GetInstanceNamespace() string {
+	return pg.Status.InstanceNamespace
```

Comment:
You could do 
```
return fmt.Sprintf("vshn-postgresql-%s", pg.GetName())
```

As `pg.Status.InstanceNamespace` is not populated during the very first reconcile.

Reply:
we have 
```
		// Wait for the next reconciliation in case instance namespace is missing
		if alertConfig.GetInstanceNamespace() == "" {
			return runtime.NewWarning(ctx, "Composite is missing instance namespace, skipping transformation")
		}
		``` 
		in alerting.go and if I set GetInstanceNamespace tests are breaking, do we want that change?

Reply:
We actually want to have the value from the InstanceNamespace as @wejdross explained. Reverted the function to return the value from the status field

Reply:
As discussed, it really doesn't matter for us if the `status.instanceNamespace` is populated or not.

The status field is generated exactly the same way via a patch in the P+T composition. It is no guarantee whatsoever that the namespace actually exists.